### PR TITLE
Th2 2339

### PR DESCRIPTION
**Pull request by Sergey-Sitnikov-Exactpro**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,12 @@
-# th2 Sailfish Utils (3.8.1)
+# th2 Sailfish Utils (3.9.0)
 
 This library contains classes to convert messages from th2 to Sailfish format and vice versa. They are used in several th2 projects to reuse Sailfish features: message comparison, codec/connect implementations, etc.
 
 ## Release Notes
 
+### 3.9.0
++ Update th2-common version for new Filter SimpleList usage
```
Comment (OptimumOpium) [inline README.md:8]: Please, add information about the previous and the new versions of the common library

**Review by OptimumOpium — CHANGES_REQUESTED** [review]
You should also update the following method in the ProtoToMessageConverter. You have added the SimpleList but it is not supported in this method

https://github.com/th2-net/th2-sailfish-utils/blob/91796c05403afef5124fd546975cff95c773c846/src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java#L167

File: src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java [hunk 2]
```diff
@@ -171,7 +171,15 @@ public IMessage fromMetadataFilter(MetadataFilter filter, String messageName) {
         IMessage message = messageFactory.createMessage(dictionaryURI, messageName);
         for (Entry<String, SimpleFilter> filterEntry : filter.getPropertyFiltersMap().entrySet()) {
             SimpleFilter propertyFilter = filterEntry.getValue();
-            message.addField(filterEntry.getKey(), toSimpleFilter(propertyFilter.getOperation(), propertyFilter.getValue()));
+            if (propertyFilter.hasSimpleList()) {
+                if (propertyFilter.getOperation() == FilterOperation.IN || propertyFilter.getOperation() == FilterOperation.NOT_IN) {
+                    message.addField(filterEntry.getKey(), new ListContainFilter(propertyFilter.getOperation(), propertyFilter.getSimpleList().getSimpleValuesList()));
+                } else {
+                    throw new IllegalArgumentException(String.format("The operation doesn't match the values {%s}, {%s}", propertyFilter.getOperation(), propertyFilter.getSimpleList()));
+                }
```
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java:179]:
```suggestion
                if (propertyFilter.getOperation() != FilterOperation.IN && propertyFilter.getOperation() != FilterOperation.NOT_IN) {
                throw new IllegalArgumentException(String.format("The operation doesn't match the values {%s}, {%s}", propertyFilter.getOperation(), propertyFilter.getSimpleList()));
                }
                message.addField(filterEntry.getKey(), new ListContainFilter(propertyFilter.getOperation(), propertyFilter.getSimpleList().getSimpleValuesList()));
```

Comment (OptimumOpium) [reply src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java:179]: Please, add a test for the new option to `ProtoToIMessageConverterWithoutDictionaryTest` class

**Review by OptimumOpium — CHANGES_REQUESTED** [review]

File: src/test/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverterWithoutDictionaryTest.java [hunk 3]
```diff
@@ -127,4 +136,38 @@ void convertsMetadataFilter() {
...
+                Arguments.of("A", StatusType.PASSED, FilterOperation.IN),
+                Arguments.of("D", StatusType.FAILED, FilterOperation.IN),
+                Arguments.of("D", StatusType.PASSED, FilterOperation.NOT_IN),
+                Arguments.of("A", StatusType.FAILED, FilterOperation.NOT_IN)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inOperationFilter")
+    void testListContainsValueFilter(String value, StatusType status, FilterOperation operation) {
```
Comment (OptimumOpium) [inline src/test/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverterWithoutDictionaryTest.java:151]: I would also add a test for negative behavior (IN with simple value)

File: src/test/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverterWithoutDictionaryTest.java [hunk 4]
```diff
@@ -127,4 +136,38 @@ void convertsMetadataFilter() {
...
+        MetadataFilter metadataFilter = MetadataFilter.newBuilder()
+                .putPropertyFilters("prop1", SimpleFilter.newBuilder()
+                        .setOperation(operation)
+                        .setSimpleList(SimpleList.newBuilder()
+                                .addAllSimpleValues(Arrays.asList("A", "B", "C")))
```
Comment (OptimumOpium) [inline src/test/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverterWithoutDictionaryTest.java:156]:
```suggestion
                                .addAllSimpleValues(List.of("A", "B", "C")))
```

**Review by OptimumOpium — APPROVED** [review]
